app/src/format.py

```diff
@@ -87,42 +92,52 @@ def format_bem_subsections(
-            else ""
-        )
-        citations_html += f"""
-        <div class="usa-accordion" id=accordion-{_accordion_id}>
-            <h4 class="usa-accordion__heading">
-                <button
-                    type="button"
-                    class="usa-accordion__button"
-                    aria-expanded="false"
-                    aria-controls="a-{_accordion_id}">
-                    {citation_number}. {chunk.document.name}
-                </button>
-            </h4>
-            <div id="a-{_accordion_id}" class="usa-accordion__content usa-prose" hidden>
-                {citation_headings}
-                {citation_body}
-                {citation_link}
-            </div>
-        </div>"""
+
+    citations_by_document = combine_citations_by_document(citation_to_numbers)
+    for document, chunk_list in citations_by_document.items():
+        for chunk in chunk_list:
+            for citation, grouped_citations in chunk.items():
+                _accordion_id += 1
+                citation_body = ""
+                citation_numbers = []
+                for citation_item in grouped_citations:
+                    for citation_number, subsection in citation_item.items():
+                        citation_numbers.append(f"{citation_number}")
+                        citation_body += f'<div>Citation #{citation_number}: </div><div class="margin-left-2 border-left-1 border-base-lighter padding-left-2">{subsection}</div>'
+
+                formatted_citation_body = to_html(replace_bem_with_link(citation_body))
+                bem_url_for_page = get_bem_url(document.name)
+                if citation.page_number:
+                    bem_url_for_page += "#page=" + str(citation.page_number)
+
+                citation_headings = (
+                    "<p>" + " → ".join(citation.headings) + "</p>" if citation.headings else ""
+                )
+                citation_link = (
+                    (
+                        f"<p><a href={bem_url_for_page!r}>Open document to page {citation.page_number}</a></p>"
+                    )
+                    if citation.page_number
+                    else ""
+                )
+            citations_html += f"""
+            <div class="usa-accordion" id=accordion-{_accordion_id}>
+                <h4 class="usa-accordion__heading">
+                    <button
+                        type="button"
+                        class="usa-accordion__button"
+                        aria-expanded="false"
+                        aria-controls="a-{_accordion_id}">
+                        {",".join(citation_numbers)}. {document.name}
+                    </button>
+                </h4>
+                <div id="a-{_accordion_id}" class="usa-accordion__content usa-prose" hidden>
+                    {citation_headings}
+                    {formatted_citation_body}
+                    {citation_link}
+                </div>
+            </div>"""
```

Comment:
I believe variations of this block appears multiple times in this file. Refactor this into a method that is called with arguments to account for the variations.